test/ConfigAggregatorTest.php

```diff
@@ -90,10 +90,69 @@ function () {
                 return ['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true];
             }
         ], $this->cacheFile);
-        $this->assertTrue(file_exists($this->cacheFile));
+        $this->assertFileExists($this->cacheFile);
         $cachedConfig = include $this->cacheFile;
-        $this->assertInternalType('array', $cachedConfig);
-        $this->assertEquals(['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true], $cachedConfig);
+
+        $this->assertIsArray($cachedConfig);
+        $this->assertSame(['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true], $cachedConfig);
+    }
+
+    public function testConfigAggregatorCanCacheConfigWithClosures()
+    {
+        new ConfigAggregator([
+                function () {
+                    return [
+                     'toUpper' => function ($input) {
+                        return strtoupper($input);
+                     },
+                     ConfigAggregator::ENABLE_CACHE => true
+                    ];
+                }
+         ], $this->cacheFile);
+
+        $this->assertFileExists($this->cacheFile);
+
+        $cachedConfig = include $this->cacheFile;
+
+        $this->assertIsCallable($cachedConfig['toUpper']);
+        $this->assertSame('FOOBAR', call_user_func($cachedConfig['toUpper'], 'foobar'));
+    }
+
+    public function testConfigAggregatorCanCacheConfigWithClosuresWithUse()
+    {
+        $prefix = 'prefix';
+        $functionWithUse = function ($input) use ($prefix) {
+            return $prefix . $input;
+        };
+        new ConfigAggregator([
+            function () use ($functionWithUse) {
+                return [
+                 'addPrefix' => $functionWithUse,
+                 ConfigAggregator::ENABLE_CACHE => true
+                ];
+            }
+         ], $this->cacheFile);
+
+        $this->assertFileExists($this->cacheFile);
+
+        $cachedConfig = include $this->cacheFile;
+
+        $this->assertIsCallable($cachedConfig['addPrefix']);
+        $this->assertSame('prefixfoobar', call_user_func($cachedConfig['addPrefix'], 'foobar'));
```

Comment:
`call_user_func()` usage should can be changed with direct call: 

```suggestion
        $this->assertSame('prefixfoobar', $cachedConfig['addPrefix']('foobar'));
```

ref https://3v4l.org/IGiDm